### PR TITLE
fix syncing screen only shows paired label after the restart

### LINKED ISSUE #21259: On the recovered device - The syncing screen only shows paired label after the restart
# Bug Report

**Follow up:** https://github.com/status-im/status-mobile/pull/21090

### Reproduction
1. Device A1 syncs with Device A.
**Actual result:**  
**Device A1**: immediately shows Device A in the list with the 'pairing' state, but the name of Device A is not displayed
**Device A:** in the settings, Device A1 is shown with the 'Unpair' state and with its name

3. On Device A, click the 'Pair and Sync' button in the sync popup.
**Device A1**: the name of the paired device appears in the list.
**Device A:** nothing will visually change on the list ,but pairing and syncing have occurred (if you send messages in 1-1 chats, they will appear on both devices).

**Expected results:**
In my opinion, both devices should change each other's states to "Pair" and visually display this to the user in the settings simultaneously after clicking the 'Pair and Sync' button.

Video from device A1: 

https://github.com/user-attachments/assets/ce7fd793-af8a-4e2d-aa5f-be73d91027f2

Screen from device A before and after click on button:
photo_2024-09-12_12-40-22
photo_2024-09-12_12-40-14

### Additional Information

- Status version: PR build #21090
- Operating System: Android, iOS

### PR DESCRIPTION
**Pull request by Parveshdhull**
fixes https://github.com/status-im/status-mobile/issues/21259
https://github.com/status-im/status-mobile/issues/21260

~status-go PR: https://github.com/status-im/status-go/pull/5869~
status-go PR: https://github.com/status-im/status-go/pull/5888

status: ready

### CODE REVIEW

File: src/legacy/status_im/pairing/core.cljs [hunk 1]
```diff
@@ -256,9 +256,10 @@
   {:events [:pairing/pair-and-sync]}
   [cofx installation-id]
   {:fx [[:json-rpc/call
-         [{:method     "wakuext_enableInstallationAndSync"
-           :params     [{:installationId installation-id}]
-           :on-success #(log/debug "successfully synced devices")}]]]})
+         [{:method      "wakuext_enableInstallationAndSync"
+           :params      [{:installationId installation-id}]
+           :js-response true
```
Comment (qfrank) [inline src/legacy/status_im/pairing/core.cljs:261]: emm.. so I missed this, this is why it not worked for me?

Comment (Parveshdhull) [reply src/legacy/status_im/pairing/core.cljs:261]: yes, `:sanitize-messages-and-process-response` expects js object

**Review by qfrank — APPROVED** [review]

**Review by seanstrom — APPROVED** [review]

**Comment by Horupa-Olena** [comment]
@Parveshdhull Hi!
Thanks for your quick fix. Please resolved conflicts and rebase this PR.

**Comment by Parveshdhull** [comment]
Thank you, @Horupa-Olena , for taking the PR for testing.

> Thanks for your quick fix.

Actually, @qfrank is the one who fixed this issue :)

> Please resolved conflicts and rebase this PR

done 👍

**Comment by qfrank** [comment]
Hi @Horupa-Olena , pls check if this PR fixed #21259 #21260 , thank you! cc @Parveshdhull

**Comment by Horupa-Olena** [comment]
@qfrank Thanks for PR! Taking it into work!

**Comment by Horupa-Olena** [comment]
@Parveshdhull @qfrank Hello! Thank you very much for your work. Everything works perfectly in both PRs #21259 #21260.

I have a question regarding the Syncing tab in Settings.
I think this is the expected result, but I need your confirmation (or correction).

**Scenario 1:**
1. Device 2 syncs with Device 1 via fallback flow.
2. Don't confirm Pair and sync in notification in AC
3. We go to Settings -> Sync on Device 2 (expected result?)
photo_2024-09-30_16-07-56

4. We go to Settings -> Sync on Device 1
photo_2024-09-30_16-07-59

**Scenario 2:**
1. Device 2 syncs with Device 1 via the main flow twice.
2. We go to Settings -> Sync on Device 1 (expected result?)
photo_2024-09-30_16-08-01

**Comment by qfrank** [comment]
> @Parveshdhull @qfrank Hello! Thank you very much for your work. Everything works perfectly in both PRs #21259 #21260.
> 
> I have a question regarding the Syncing tab in Settings. I think this is the expected result, but I need your confirmation (or correction).
> 
> **Scenario 1:**
> 
> 1. Device 2 syncs with Device 1 via fallback flow.
> 2. Don't confirm Pair and sync in notification in AC
> 3. We go to Settings -> Sync on Device 2 (expected result?)
>    photo_2024-09-30_16-07-56
> 4. We go to Settings -> Sync on Device 1
>    photo_2024-09-30_16-07-59
> 
> **Scenario 2:**
> 
> 1. Device 2 syncs with Device 1 via the main flow twice.
> 2. We go to Settings -> Sync on Device 1 (expected result?)
>    photo_2024-09-30_16-08-01

yeah, both are expected! @Horupa-Olena

**Comment by Horupa-Olena** [comment]
@qfrank Great! I don't see any issue, you can merge this PR.

**Comment by qfrank** [comment]
> @qfrank Great! I don't see any issue, you can merge this PR.

waiting for approval of the corresponding status-go [PR](https://github.com/status-im/status-go/pull/5888) @status-im/status-go-guild

**Comment by churik** [comment]
@qfrank please, merge the PR along with https://github.com/status-im/status-go/pull/5888 (I believe it is ready already)

We plan to check it one more time in nightly before release cut.

Thanks!!!

**Comment by qfrank** [comment]
> @qfrank please, merge the PR along with [status-im/status-go#5888](https://github.com/status-im/status-go/pull/5888) (I believe it is ready already)
> 
> We plan to check it one more time in nightly before release cut.
> 
> Thanks!!!

I wish I could ... An [error](https://ci.status.im/blue/organizations/jenkins/status-go%2Fprs%2Ftests/detail/PR-5888/12/pipeline) unrelated to the PR prevented it from merging. I'm waiting for the [PR](https://github.com/status-im/status-go/pull/5922) to be merged first as it will resolve the error. It should be merged days before if there's no such error @churik cc @igor-sirotin

**Comment by qfrank** [comment]
going to merge it when the green merge button is ready @churik @Horupa-Olena

**Pull request merged**